packages/opentelemetry/lib/tracing/tracerProvider.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable no-unused-vars */
+
+import { getBatchSpanProcessor } from './spanProcessor';
+import { buildExporter, AVAILABLE_EXPORTERS } from './exporter';
+import { NodeTracerProvider } from '@opentelemetry/node';
+import { registerInstrumentations, InstrumentationOption } from '@opentelemetry/instrumentation';
+import { ServerInstrumentation } from './serverInstrumenation';
+import { diag, DiagConsoleLogger, DiagLogLevel } from '@opentelemetry/api';
+import { SpanProcessor, } from '@opentelemetry/tracing';
+
+//Delegate
+class TracerProvider {
+  constructor () {
+    this.provider = new NodeTracerProvider();
+    this.exporters = {};
+    this.init();
+  }
+
+  init () {
+    diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.ALL);
+    this.provider.register();
+
+    const consoleExporter = buildExporter(AVAILABLE_EXPORTERS.CONSOLE);
+    const spanProcessor = getBatchSpanProcessor(consoleExporter);
+    this.addSpanProcessor(spanProcessor);
+    this.exporters[AVAILABLE_EXPORTERS.CONSOLE] = consoleExporter;
+
+    this._serverInstrumentation = new ServerInstrumentation();
+    registerInstrumentations({
+      instrumentation: this._serverInstrumentation.instrumentations,
+      tracerProvider: this.provider
+    });
+    this._active = true;
+    this.currentConfig = {
+      active: this._active,
+      current_exporters: [AVAILABLE_EXPORTERS.CONSOLE],
+      exporters:
+        [
+          {
+            exporter_type: 'console',
+            config: {}
+          }
+        ]
+    };
+  }
+
+  getCurrentConfig () {
+    return JSON.parse(JSON.stringify(this.currentConfig));
```

Comment:
is it necessary to perform this double transformation?

Reply:
this is just to return not the reference but a clone object of the same

Reply:
you could you lodash's cloneDeep method for that